sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -161,7 +161,10 @@ std::unique_ptr<RawResponse> CurlTransport::Send(Request& request, Context const
   Log::Write(Logger::Level::Verbose, LogMsgPrefix + "Creating a new session.");
 
   auto session = std::make_unique<CurlSession>(
-      request, CurlConnectionPool::GetCurlConnection(request, m_options), m_options.HttpKeepAlive);
```

Comment:
This is also not correct.
Revert this change. The `CurlConnectionPool` is a private/detail implementation.  There's no need  or benefit of renaming this method.

In this case, the connection might be already in the pool and `extract` might sound OK, but if there is not a connection in the pool, a new connection will be created.  The ConnectionPool is a `facade` for getting or creating a connection for the consumer.

Reply:
I'll rename it to ExtractOrCreate()